uaclient/cli/fix.py

```diff
@@ -52,6 +104,747 @@ def fix_parser(parser):
+        token = input("> ")
+        return _run_ua_attach(cfg, token)
+
+    return True
+
+
+def _format_unfixed_packages_msg(unfixed_pkgs: List[str]) -> str:
+    """Format the list of unfixed packages into an message.
+
+    :returns: A string containing the message output for the unfixed
+              packages.
+    """
+    num_pkgs_unfixed = len(unfixed_pkgs)
+    return textwrap.fill(
+        messages.SECURITY_PKG_STILL_AFFECTED.format(
+            num_pkgs=num_pkgs_unfixed,
+            s="s" if num_pkgs_unfixed > 1 else "",
+            verb="are" if num_pkgs_unfixed > 1 else "is",
+            pkgs=", ".join(sorted(unfixed_pkgs)),
+        ).msg,
+        width=PRINT_WRAP_WIDTH,
+        subsequent_indent="    ",
+    )
+
+
+def _check_subscription_is_expired(cfg: UAConfig, dry_run: bool) -> bool:
+    """Check if the Ubuntu Pro subscription is expired.
+
+    :returns: True if subscription is expired and not renewed.
+    """
+    contract_expiry_status = get_contract_expiry_status(cfg)
+    if contract_expiry_status[0] == ContractExpiryStatus.EXPIRED:
+        if dry_run:
+            print(messages.SECURITY_DRY_RUN_UA_EXPIRED_SUBSCRIPTION)
+            return False
+        return True
+
+    return False
+
+
+def _prompt_for_new_token(cfg: UAConfig) -> bool:
+    """Prompt for attach a new subscription token to the user.
+
+    :return: True if attach performed.
+    """
+    import argparse
+
+    from uaclient import cli
+
+    _inform_ubuntu_pro_existence_if_applicable()
+    print(messages.SECURITY_UPDATE_NOT_INSTALLED_EXPIRED)
+    choice = util.prompt_choices(
+        "Choose: [R]enew your subscription (at {}) [C]ancel".format(
+            PRO_HOME_PAGE
+        ),
+        valid_choices=["r", "c"],
+    )
+    if choice == "r":
+        print(messages.PROMPT_EXPIRED_ENTER_TOKEN)
+        token = input("> ")
+        print(colorize_commands([["pro", "detach"]]))
+        cli.action_detach(
+            argparse.Namespace(assume_yes=True, format="cli"), cfg
+        )
+        return _run_ua_attach(cfg, token)
+
+    return False
+
+
+def _prompt_for_enable(cfg: UAConfig, service: str) -> bool:
+    """Prompt for enable a pro service.
+
+    :return: True if enable performed.
+    """
+    import argparse
+
+    from uaclient import cli
+
+    print(messages.SECURITY_SERVICE_DISABLED.format(service=service))
+    choice = util.prompt_choices(
+        "Choose: [E]nable {} [C]ancel".format(service),
+        valid_choices=["e", "c"],
+    )
+
+    if choice == "e":
+        print(colorize_commands([["pro", "enable", service]]))
+        return bool(
+            0
+            == cli.action_enable(
+                argparse.Namespace(
+                    service=[service],
+                    assume_yes=True,
+                    beta=False,
+                    format="cli",
+                    access_only=False,
+                ),
+                cfg,
+            )
+        )
```

Comment:
This might be a good opportunity to change this to use `actions.enable_entitlement_by_name`